src/families/bitcoin/wallet-btc/crypto/bitcoin.ts

```diff
@@ -140,7 +141,12 @@ class Bitcoin extends Base {
   }
 
   private tryBech32(address: string): boolean {
-    const result = fromBech32(address);
+    let result;
+    try {
+      result = fromBech32(address);
+    } catch {
+      return false;
```

Comment:
I think it's wrong to return false here. If you look at Bitcoin.validateAddress, it expects an exception if bech32 decoding fails, so that it can try base58 instead:

```
    try {
      // This prefix check is to avoid returning false in cases where a valid base58 address also happens
      // to be a valid bech32(m) string (but invalid segwit address).
      if (address.toLowerCase().startsWith(`${this.network.bech32}1`)) {
        return this.tryBech32(address);
      }
    } catch {
      /* Try base58 instead */
    }
```
I'm surprised that this wasn't caught by tests. I'm almost certain it would be caught by "toOutputScript p2sh and p2pkh work as expected" in bitcoin.test.ts.

Reply:
Ok, it's only when the address starts with bc1... that tryBech32 is called, so it's only in extreme (weird) edge cases this is a problem. But I still think throwing is the correct thing to do as my comment suggests.